src/Microsoft.Sbom.DotNetTool/GenerationService.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Sbom.Api;
+using Microsoft.Sbom.Api.Exceptions;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Api.Workflows;
+
+namespace Microsoft.Sbom.Tool
+{
+    public class GenerationService : IHostedService
+    {
+        private readonly IWorkflow<SBOMGenerationWorkflow> generationWorkflow;
+        private readonly IRecorder recorder;
+        private readonly IHostApplicationLifetime hostApplicationLifetime;
+
+        public GenerationService(
+            IWorkflow<SBOMGenerationWorkflow> generationWorkflow,
+            IRecorder recorder,
+            IHostApplicationLifetime hostApplicationLifetime)
+        {
+            this.generationWorkflow = generationWorkflow;
+            this.recorder = recorder;
+            this.hostApplicationLifetime = hostApplicationLifetime;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var result = await generationWorkflow.RunAsync();
+                await recorder.FinalizeAndLogTelemetryAsync();
+                Environment.ExitCode = result ? (int)ExitCode.Success : (int)ExitCode.GeneralError;
+            }
+            catch (AccessDeniedValidationArgException e)
+            {
+                var message = e.InnerException != null ? e.InnerException.Message : e.Message;
+                Console.WriteLine($"Encountered error while running ManifestTool generation workflow. Error: {message}");
```

Comment:
nit: `ManifestTool` -> `SBOM Tool`

Reply:
I know this stuff is irrelevant to the main idea of PR, but this huge PR actually helps notice some stuff that would be great to address now or in the future :)